src/promptix/core/base_refactored.py

```
@@ -0,0 +1,254 @@
+                available_prompts = []
+            raise PromptNotFoundError(
+                prompt_name=prompt_template,
+                available_prompts=available_prompts
+            ) from err
+
+        versions = prompt_data.get("versions", {})
+        version_data = self._version_manager.get_version_data(versions, version, prompt_template)
+        
+        # Build the model configuration
+        return self._model_config_builder.build_model_config(
+            system_message=system_message,
```

Ruff (0.12.2)

Undefined name `system_message`

(F821)